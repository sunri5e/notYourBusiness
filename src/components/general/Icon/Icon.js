// Ayasdi Inc. Copyright 2017 - all rights reserved.

import React from 'react';
import {
  string, bool, shape, oneOf,
} from 'prop-types';

const Icon = ({
  icon, className, custom, size, style, fontType,
}) => (
  custom ? (
    <svg className={`app-icon fa fa-${icon} ${className} ${size ? `app-icon__size-${size}` : ''}`} style={style}>
      <use xlinkHref={`#fa-${icon}`} />
    </svg>
  ) : (
    <span className={`app-icon ${className} ${size ? `app-icon__size-${size}` : ''}`} style={style}>
      <i className={`${fontType || 'fa'} fa-${icon}`} />
    </span>
  )
);

Icon.propTypes = {
  icon: string.isRequired,
  /**
   * Type of Font Awesome.
   */
  fontType: oneOf(['', 'far']),
  className: string,
  /**
   * Icon size. Empty mean default/medium.
   */
  size: oneOf(['', 'xxl', 'xl', 'l', 's', 'xs']),
  custom: bool,
  style: shape(),
};
Icon.defaultProps = {
  fontType: '',
  className: '',
  size: '',
  custom: false,
  style: {},
};

export default Icon;
