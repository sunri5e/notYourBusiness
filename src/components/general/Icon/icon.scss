// Ayasdi Inc. Copyright 2017 - all rights reserved.
.app-icon {
  fill: currentColor;
  color: currentColor;
  line-height: 1;
  text-align: center;
  display: inline-block;
  width: $size-base * 3.5;
  height: $size-base * 3.5;
  font-size: $size-base * 3.5;
  &__size-l {
    width: $size-base * 4;
    height: $size-base * 4;
    font-size: $size-base * 4;
  }
  &__size-xl {
    width: $size-base * 6;
    height: $size-base * 6;
    font-size: $size-base * 6;
  }
  &__size-xxl {
    width: $size-base * 8;
    height: $size-base * 8;
    font-size: $size-base * 8;
  }
  &__size-s {
    width: $size-base * 3;
    height: $size-base * 3;
    font-size: $size-base * 3;
  }
  &__size-xs {
    width: $size-base * 2;
    height: $size-base * 2;
    font-size: $size-base * 2;
  }
}

// Prevent IE freezing when clicking icons
svg.app-icon use {
  pointer-events: none;
}
