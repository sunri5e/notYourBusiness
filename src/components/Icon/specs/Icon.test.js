// Ayasdi Inc. Copyright 2017 - all rights reserved.

import React from 'react';
import { shallow } from 'enzyme';
import Icon from '../Icon';

describe('<Icon />', () => {
  const wrapper = shallow(<Icon icon="someicon" />);

  it('adds an icon svg with the proper id', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
