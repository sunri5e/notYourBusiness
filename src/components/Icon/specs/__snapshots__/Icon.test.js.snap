// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> adds an icon svg with the proper id 1`] = `
<span
  className="app-icon  "
  style={Object {}}
>
  <i
    className="fa fa-someicon"
  />
</span>
`;
